spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/paths/BenchmarkPaths.java

```diff
@@ -59,13 +60,15 @@ public String parquetDir(int scale) {
         return String.join(File.separator, rootDataDir(scale), "raw_parquet");
     }
 
-    public String experimentResultLocation(int scale, String queryName) {
+    public String experimentResultLocation(QuerySessionIdentifier identifier, int attempt) {
         return String.join(
                 File.separator,
                 "experiments_query_results",
                 experimentName,
-                String.format("scale=%d", scale),
-                StringUtils.removeEnd(queryName, ".sql"));
+                String.format("scale=%d", identifier.scale()),
+                StringUtils.removeEnd(identifier.queryName(), ".sql"),
+                String.format("session=%s", identifier.session()),
```

Comment:
should we do the same for experimentCorrectnessHashesRoot? (I'm still not sure where the correctness hashes are actually from :D)

Reply:
we could add `session`... from my understanding of the code, we just write the table hash on the first iteration, so on subsequent iterations we compare for equality (or like, if we pick our paths so that they're the same across runs, then it gets written on first iteration of first run and read from there forever after)